spec/spec_helper.rb

```diff
@@ -173,11 +177,14 @@
 # To allow for leaky threads to be traced
 # back to their creation point.
 module DatadogThreadDebugger
-  def initialize(*args)
+  # DEV: we have to use an explicit `block`, argument
+  # instead of the implicit `yield` call, as calling
+  # `yield` here crashes the Ruby VM in Ruby < 2.2.
+  def initialize(*args, &block)
     caller_ = caller
     wrapped = lambda do |*thread_args|
       Thread.current.instance_variable_set(:@caller, caller_)
-      yield(*thread_args)
+      block.call(*thread_args) # rubocop:disable Performance/RedundantBlockCall
     end
     wrapped.ruby2_keywords if wrapped.respond_to?(:ruby2_keywords, true)
```

Comment:
I like how this method is like the history of Ruby bugs and features. "Let's just wrap it" turns into "oh but this crashes on old rubies, and we need to use correct delegation on modern rubies...". Ahh computers...